src/zope/app/publisher/xmlrpc/testing.py

```diff
@@ -76,10 +76,10 @@ def request(self, host, handler, request_body, verbose=0):
                 headers)
 
         body = response.getBody()
-        if not isinstance(body, str):
+        if not isinstance(errmsg, bytes):
```

Comment:
👍 for some encoding here ('ascii' is consistent with the rest of the file), but I worry now that there's still some non-explicit conversions (Python 2) or TypeErrors (Python 3) lurking. Maybe be explicit about everything?
```suggestion
        body = body if isinstance(body, bytes) else body.encode('latin-1')
        errmsg = errmsg if isinstance(errmsg, bytes) else errmsg.encode('ascii') # HTTP response lines are ASCII
```

Reply:
Nice, I am going to use your suggestion but have to wrap it to several lines.